samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -29,20 +44,71 @@ static const struct bt_data ad[] = {
 			  0x12, 0x18,	/* HID Service */
 #endif
 			  0x0f, 0x18),	/* Battery Service */
-	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
-};
 
-static const struct bt_data sd[] = {
+#if CONFIG_DESKTOP_BLE_SWIFT_PAIR
+	BT_DATA(BT_DATA_NAME_SHORTENED, DEVICE_NAME, DEVICE_NAME_SHORT_LEN),
+	BT_DATA_BYTES(BT_DATA_MANUFACTURER_DATA,
+			  0x06, 0x00,	/* Microsoft Vendor ID */
+			  0x03,		/* Microsoft Beacon ID */
+			  0x00,		/* Microsoft Beacon Sub Scenario */
+			  0x80),	/* Reserved RSSI Byte */
+#else
 	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+#endif
 };
 
+static const struct bt_data sd[] = {};
+
 static bool bonds_initialized;
 static bool bonds_remove;
 
+#if CONFIG_DESKTOP_BLE_SWIFT_PAIR
+static struct k_delayed_work adv_update;
+static bool vendor_section_remove;
+
+static void ble_adv_update_fn(struct k_work *work)
+{
+	int err;
+	if (vendor_section_remove) {
+		LOG_INF("Removing vendor section");
+		err = bt_le_adv_update_data(
+			ad, (ARRAY_SIZE(ad) - SWIFT_PAIR_SECTION_SIZE),
+			sd, ARRAY_SIZE(sd));
+	} else {
+		LOG_INF("Switch to normal cadence advertising");
+		adv_param.interval_min = BT_GAP_ADV_FAST_INT_MIN_2;
+		adv_param.interval_max = BT_GAP_ADV_FAST_INT_MAX_2;
+
+		err = bt_le_adv_stop();
+
+		if (!err) {
+			bt_le_adv_start(&adv_param, ad, ARRAY_SIZE(ad),
+					sd, ARRAY_SIZE(sd));
+		}
+
+		if (err) {
+			LOG_ERR("Failed to restart advertising (err %d)", err);
+			sys_reboot(SYS_REBOOT_WARM);
```

Comment:
please do not warm reboot the system here (also remove the original code). instead set module state to error - we will handle error within error_module (to be implemented)

Reply:
removed reboot from this module.